app/ui/browser-modern/sagas/session-sagas.js

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import { takeLatest, throttle } from 'redux-saga';
+import { race, call, take, put, select } from 'redux-saga/effects';
+
+import { ipcRenderer } from '../../../shared/electron';
+import { wrapped } from './helpers';
+import { getWindowId } from '../selectors/ui';
+import { serializeAppState } from '../util/session-util';
+import * as ActionTypes from '../constants/action-types';
+import * as EffectTypes from '../constants/effect-types';
+import * as SessionEffects from '../actions/session-effects';
+
+const THROTTLE_DURATION = 1000; // ms
```

Comment:
nit: comment on what this really means.  Is it that we'll never write more than once every second?